src/ClientModel/CaptureData.cpp

```diff
@@ -93,7 +93,15 @@ void CaptureData::UpdateFunctionStats(uint64_t instrumented_function_id, uint64_
   FunctionStats& stats = functions_stats_[instrumented_function_id];
   stats.set_count(stats.count() + 1);
   stats.set_total_time_ns(stats.total_time_ns() + elapsed_nanos);
+  uint64_t old_avg = stats.average_time_ns();
   stats.set_average_time_ns(stats.total_time_ns() / stats.count());
+  uint64_t old_variance = stats.std_dev_ns() * stats.std_dev_ns();
+  // variance(N) = ( (N-1)*variance(N-1) + (x-avg(N))*(x-avg(N-1)) ) / N
+  double variance = ((stats.count() - 1) * old_variance +
```

Comment:
By recomputing `old_variance` as the square of the old std_dev, I think we risk accumulating more error than if we were to store the variance for the next iteration : `(sqrt(value))^2 = value + error`. Could we store it instead?

There would still be error accumulation if we stored the variance, but probably less. I wonder if we should also recompute the std_dev using all the final values once capture stops. I'd be curious to see if there is a noticeable difference with the std_dev value we get by this method.

Reply:
Yeah I was going to bring up if we should store variance. I added that part.

> using all the final values

Do you mean store every elapsed_nanos that we get and recompute the standard deviation at the end? We'd be storing a lot more

Reply:
I mean that we might want to recompute `std_dev` by iterating over all the recorded timers once on capture stop so that we get the most precise value possible, one which doesn't suffer from error accumulation. This could be done in another PR, but we should probably open a bug for it.